spec/models/patient_spec.rb

```diff
@@ -206,5 +206,60 @@
       end
     end
   end
+
+  context '.discard_data' do
+    before do
+      create_list(:prescription_drug, 2, patient: patient)
+      create(:medical_history, patient: patient)
+      create_list(:appointment, 2, patient: patient)
+      bps = create_list(:blood_pressure, 2, patient: patient)
+      sugars = create_list(:blood_sugar, 2, patient: patient)
+      (bps + sugars).each do |record|
+        create(:encounter,
+               :with_observables,
+               patient: patient,
+               observable: record,
+               facility: patient.registration_facility)
+      end
+    end
+
+    it "should discard a patient's address" do
+      patient.discard_data
+      expect(patient.address.discarded?).to be true
+      expect(Address.find_by(id: patient.address.id)).to be nil
+    end
+    it "should discard a patient's appointments" do
+      expect { patient.discard_data }.to change { patient.appointments.count }.by(-2)
```

Comment:
Isn't change `to` without a `from` just `eq`? Would prefer to use `by` so that we know the data creation in the setup is actually working like we expect i.e., the records were created and then discarded as opposed to not getting created at all.